### PR TITLE
Issue 2858

### PR DESCRIPTION
**Pull request by MCGallaspy**
Solves issues #2858, #2884, #2865, #2881, and #2869. (And possibly a few other issues I opened along the way.)

### CODE REVIEW

**Comment by MCGallaspy** [comment]
And #2864.

File: kalite/distributed/static/js/distributed/topics/router.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ ChannelRouter = Backbone.Router.extend({
 
     routes: {
         "":   "navigate_default_channel",
-        ":channel/(*splat)":    "navigate_channel"
+        ":channel/(*splat)":    "navigate_channel",
+        "/(.*)/": "trigger_navigation_callback"
```
Comment (rtibbles) [inline kalite/distributed/static/js/distributed/topics/router.js]: Might be good to comment what the purpose of this is.

File: kalite/main/api_resources.py [hunk 2]
```diff
@@ -269,6 +270,7 @@ def __init__(self, lang_code="en", **kwargs):
         self.selected_language = lang_code
         if self.description == "None":
             self.description = ""
+        self.source = settings.CHANNEL
```
Comment (rtibbles) [inline kalite/main/api_resources.py]: Can you add a TODO (BLOCKER) to change this for 0.14? We will need to make this happen dynamically when we have multiple channels active at once.

**Comment by MCGallaspy** [comment]
Tried running the tests on my own machine, I'm also hanging on this test:
`test_user_interface_teacher (kalite.control_panel.tests.control_panel.CSVExportBrowserTests)`
I'll look in to it.

**Comment by MCGallaspy** [comment]
Hanging on a test again (`test_points_update_after_backbone_navigation`), which is a bit disturbing because it doesn't hang on that test on my machine... non-deterministic browser tests?

**Comment by MCGallaspy** [comment]
I'm deeply suspicious of this, as the indicated failed tests run fine on my machine, and further my own machine fails tests which are passing here. I think the deeper issue is with our testing framework, and the fact that variable server response times are causing expected DOM elements (or in my case, certain javascript objects) to be missing or not populated in the expected way. Thoughts on how to proceed?

**Comment by aronasorman** [comment]
Talked with @MCGallaspy. We decided to just add `skipIf`s since the tests are failing for no reason, and it's the test suite's fault, not our code. We'll revisit this once we start refactoring the whole testing infrastructure.

**Comment by rtibbles** [comment]
I agree - there is a limit to how robust we can make our client side testing by doing it through Selenium in this way. Roll on #2870!

**Comment by MCGallaspy** [comment]
Okay, maybe for now if a test fails on Travis we can just get someone to check it out manually before merging if we suspect that it's a failure of the test suite and not an actual bug?

**Comment by rtibbles** [comment]
Yeah, just adding that you ran the test manually, and setting it to skip if travis is probably best. If you want to do that with these failing tests, we can get this PR merged!

**Comment by rtibbles** [comment]
All comments addressed. Tests pass. Merging.

**Pull request merged**